network/common/src/main/java/org/apache/spark/network/server/RpcHandler.java

```diff
@@ -48,10 +54,40 @@ public abstract void receive(
   public abstract StreamManager getStreamManager();
 
   /**
+   * Receives an RPC message that does not expect a reply. The default implementation will
+   * call "{@link receive(TransportClient, byte[], RpcResponseCallback}" and log a warning if
+   * any of the callback methods are called.
+   *
+   * @param client A channel client which enables the handler to make requests back to the sender
+   *               of this RPC. This will always be the exact same object for a particular channel.
+   * @param message The serialized bytes of the RPC.
+   */
+  public void receive(TransportClient client, byte[] message) {
+    receive(client, message, ONE_WAY_CALLBACK);
+  }
+
+  /**
    * Invoked when the connection associated with the given client has been invalidated.
    * No further requests will come from this client.
    */
   public void connectionTerminated(TransportClient client) { }
 
   public void exceptionCaught(Throwable cause, TransportClient client) { }
+
+  private static class OneWayRpcCallback implements RpcResponseCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(OneWayRpcCallback.class);
+
+    @Override
+    public void onSuccess(byte[] response) {
+      logger.warn("Response provided for one-way RPC.");
```

Comment:
`NettyRpcEnv` is the only user of `TransportClient.send(byte[])`. You'll only get these logs if:

- You do not override `receive(TransportClient client, byte[] message)`
- Your implementation of the other `receive()` method sets a response for one-way messages.

I could just as well just make `receive(TransportClient client, byte[] message)` throw an exception instead of having this default handler.